### DISCUSSION TITLE
How does the Leaflet map prevent browser touch/scroll behaviour?

### DISCUSSION THREAD

**Discussion by wasong** [Leaflet/Leaflet#8129] [Q&A] [answered]
I noticed on mobile browsers that a fullscreen leaflet map prevents the user from triggering browser scroll events such as pulling down to refresh, or swiping from the left edge to go back a page.

How has leaflet achieved this?

**Comment by Falke-Design** [comment]
You can look into the [TouchZoom](https://github.com/Leaflet/Leaflet/blob/main/src/map/handler/Map.TouchZoom.js) file

**Comment by IvanSanchez** [comment, marked as answer]
It all comes down to https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault, via calling `L.DomEvent.preventDefault()`... https://github.com/Leaflet/Leaflet/blob/fb0b2a169ebde77a8bed1715bb07d436b60328a2/src/dom/DomEvent.js#L205-L217

...from different places, e.g. ...

https://github.com/Leaflet/Leaflet/blob/fb0b2a169ebde77a8bed1715bb07d436b60328a2/src/map/handler/Map.TouchZoom.js#L103

...or also...

https://github.com/Leaflet/Leaflet/blob/fb0b2a169ebde77a8bed1715bb07d436b60328a2/src/map/handler/Map.ScrollWheelZoom.js#L59

...which comes to the same thing since...

https://github.com/Leaflet/Leaflet/blob/f4e7af1dcdb71fd1ce706fe7a7da086a85067496/src/dom/DomEvent.js#L219-L225
